src/runtime/GCStruct.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022-present Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "Walrus.h"
+
+#include "GCStruct.h"
+#include "runtime/TypeStore.h"
+
+#ifdef ENABLE_GC
+#include "GCUtil.h"
+#endif /* ENABLE_GC */
+
+namespace Walrus {
+
+GCStruct* GCStruct::structNew(const StructType* type, ByteCodeStackOffset* offsets, uint8_t* bp)
+{
+#ifdef ENABLE_GC
+    GCStruct* result = reinterpret_cast<GCStruct*>(GC_malloc(type->structSize()));
```

Comment:
The `GC_make_descriptor` could improve this data with marking the references in the structure. However, I could not find a way to delete the descriptor when all data is freed. Is it possible to deleting the descriptor?

Reply:
Yes, the descriptor is shared. However, when all references to a (recursive) type is deleted (all modules who used it and all allocated instances are released) the type descriptor is deleted as well. The code uses a simple reference counting.

I think we can do this in a follow-up patch, but would be good to do it. Could you tell me how the mark callback works?

Reply:
You can refer https://github.com/Samsung/escargot/blob/52bbc7a9bce5c248bfcd884854d21a7495b1e82e/src/heap/CustomAllocator.cpp

Reply:
I have replaced the macros